beekeeper-api/src/main/java/com/expediagroup/beekeeper/api/service/HousekeepingEntityService.java

```diff
@@ -13,11 +13,18 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.expediagroup.beekeeper.api;
+package com.expediagroup.beekeeper.api.service;
 
-import org.springframework.web.bind.annotation.RestController;
 
-@RestController
-public class BeekeeperController {
-  
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+
+import com.expediagroup.beekeeper.api.response.HousekeepingMetadataResponse;
+import com.expediagroup.beekeeper.core.model.HousekeepingMetadata;
+
+public interface HousekeepingEntityService<HousekeepingEntity> {
```

Comment:
Why do you have a generic in there - `<HousekeepingEntity>`?

Reply:
Because when I'll add the path endpoint, this will have to be generic so that I can have this method: `Page<HousekeepingPathResponse> getAllPaths()`